packages/api/amplify_api/test/ws/web_socket_connection_test.dart

```diff
@@ -36,78 +44,153 @@ void main() {
+
+      WebSocketConnection.httpClientOverride = mockClient;
+      WebSocketConnection.connectivityOverride = connectivity;
+      WebSocketConnection.webSocketFactoryOverride = getMockWebSocketChannel;
+
+      hubEventsController = StreamController.broadcast();
+      hubEvents = hubEventsController.stream;
+      Amplify.Hub.listen(HubChannel.Api, hubEventsController.add);
+    });
+    tearDown(() async {
+      connection.close();
+      Amplify.Hub.close();
+      await hubEventsController.close();
+    });
+    tearDownAll(() {});
+
+    test('should init a connection & call onEstablishCallback', () async {
+      getWebSocketConnection().subscribe(
+        subscriptionRequest,
+        expectAsync0(() {}),
       );
+
       expect(
-        connection.lastSentMessage?.messageType,
-        MessageType.connectionInit,
+        hubEvents,
+        emitsInOrder(
+          [
+            connectingHubEvent,
+            connectedHubEvent,
+          ],
+        ),
       );
-    });
 
-    test('subscribe() should initialize the connection and call onEstablished',
-        () async {
-      connection.subscribe(subscriptionRequest, expectAsync0(() {}));
-      await expectLater(connection.ready, completes);
-    });
-
-    test(
-        'subscribe() should send SubscriptionRegistrationMessage with authorized payload correctly serialized',
-        () async {
-      await connection.init();
-      await connection.ready;
-      final establishedCompleter = Completer<void>();
-      connection.subscribe(subscriptionRequest, establishedCompleter.complete);
-      await establishedCompleter.future;
-
-      final lastMessage = connection.lastSentMessage;
-      expect(lastMessage?.messageType, MessageType.start);
-      final payloadJson = lastMessage?.payload?.toJson();
-      final apiKeyFromPayload =
-          // ignore: avoid_dynamic_calls
-          payloadJson?['extensions']['authorization'][xApiKey];
-      expect(apiKeyFromPayload, testApiKeyConfig.apiKey);
+      await assertWebSocketConnected(connection, subscriptionRequest.id);
     });
 
     test('subscribe() should return a subscription stream', () async {
-      final subscription = connection.subscribe(
+      final subscription = getWebSocketConnection().subscribe(
         subscriptionRequest,
-        null,
+        () {},
       );
 
-      late StreamSubscription<GraphQLResponse<String>> streamSub;
-      streamSub = subscription.listen(
+      await assertWebSocketConnected(connection, subscriptionRequest.id);
+
+      final streamSub = subscription.listen(
         expectAsync1((event) {
           expect(event.data, json.encode(mockSubscriptionData));
-          streamSub.cancel();
         }),
       );
+
+      connection.channel!.sink.add(jsonEncode(mockSubscriptionEvent));
+
+      addTearDown(streamSub.cancel);
     });
 
+    // todo(equartey): Implement reconnection logic tests
+    // test('should reconnect when network turns on/off', () async {
```

Comment:
nit: I don't personally like having a bunch of commented code, so would suggest just remove this and leaving the TODO comment. Otherwise could uncomment the tests and skip them.